src/qibo/quantum_info/metrics.py

```diff
@@ -134,8 +134,34 @@ def trace_distance(state, target, check_hermitian: bool = False, backend=None):
     return distance
 
 
+def hilbert_schmidt_inner_product(operator_A, operator_B, backend=None):
+    """Calculate the Hilbert-Schmidt inner product between two operators.
+
+    Given two operators :math:`A, \\, B \\in \\mathcal{H}`, the Hilbert-Schmidt
+    inner product between the two is given by
+
+    .. math::
+        \\braket{A, \\, B}_{\\text{HS}} = \\text{tr}\\left(A^{\\dagger} \\, B\\right) \\, .
+
+    Args:
+        operator_A (ndarray): operator :math:`A`.
+        operator_B (ndarray): operator :math:`B`.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
+            in the execution. If ``None``, it uses :class:`qibo.backends.GlobalBackend`.
+            Defaults to ``None``.
+
+    Returns:
+        float: Hilbert-Schmidt inner product :math:`\\braket{A, \\, B}_{\\text{HS}}`.
+    """
+    backend = _check_backend(backend)
+
+    inner_product = backend.np.trace(backend.np.conj(operator_A.T) @ operator_B)
+
+    return float(backend.np.real(inner_product))
```

Comment:
Is there a reason why we return a float instead of a native array of the backend? I'd say it would be better to have an `array(1.)` or a `tensor(1.)` rather than just `1.`.